### PR TITLE
Extensions and refactoring

### PR DESCRIPTION
**Pull request by tomasherman**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,18 +1,13 @@
-[![Build Status](https://travis-ci.org/avast/datadog-scala-metrics.svg?branch=master)](https://travis-ci.org/avast/datadog-scala-metrics) [ ![Download](https://api.bintray.com/packages/avast/maven/datadog-scala-metrics/images/download.svg) ](https://bintray.com/avast/maven/datadog-scala-metrics/_latestVersion)
+# Datadog Scala Metrics [![Build Status](https://travis-ci.org/avast/datadog-scala-metrics.svg?branch=master)](https://travis-ci.org/avast/datadog-scala-metrics) [![Download](https://api.bintray.com/packages/avast/maven/datadog-scala-metrics/images/download.svg)](https://bintray.com/avast/maven/datadog-scala-metrics/_latestVersion) <img height="40" src="https://typelevel.org/cats/img/cats-badge-tiny.png" align="right"/>
```
Comment (hanny24) [inline docs/README.md]: Why don't you use markdown syntax for images?

File: docs/README.md [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
 
-# Datadog Scala Metrics
 
-Thin wrapper around datadog metrics that aims to provide idiomatic API for purely functional Scala.
+Toolkit for monitoring applications written in functional Scala with Datadog.
```
Comment (hanny24) [inline docs/README.md]: with Datadog  => using Datadog

File: docs/userguide.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+### Timers
+Timers are great. And with our API, they are even better. Because we are living in functional code, we expect you to provide us with `F[_]: Sync` and we will time how long execution takes, and tag it with whether it succeded and if it failed, which class of exception was thrown.
+
+```scala mdoc
+
+factoryResource.use { factory =>
+    val timer = factory.timer("request-latency")
+
+    timer.time(IO.pure(println("success"))) // marked as success
```
Comment (hanny24) [inline docs/userguide.md]: marked => tagged ?

File: docs/README.md (hunk 1 above)
Comment (tomasherman) [reply docs/README.md]: it's because of the height and align params that make this look a lot nicer

**Pull request merged**